utilities/virt.py

```diff
@@ -1573,16 +1573,20 @@ def get_rhel_os_dict(rhel_version):
     raise KeyError(f"Failed to extract {rhel_version} from system_rhel_os_matrix")
 
 
-def assert_vm_not_error_status(vm: VirtualMachineForTests) -> None:
-    status = vm.instance.get("status")
-    printable_status = status.get("printableStatus")
-    error_list = VM_ERROR_STATUSES.copy()
-    vm_devices = vm.instance.spec.template.spec.domain.devices
-    if vm_devices.gpus:
-        error_list.remove(VirtualMachine.Status.ERROR_UNSCHEDULABLE)
-    assert printable_status not in error_list, (
-        f"VM {vm.name} error printable status: {printable_status}\nVM status:\n{status}"
-    )
+def assert_vm_not_error_status(vm: VirtualMachineForTests, timeout: int = TIMEOUT_5SEC) -> None:
+    for status in TimeoutSampler(wait_timeout=timeout, sleep=TIMEOUT_1SEC, func=lambda: vm.instance.get("status", {})):
+        if status:
+            printable_status = status.get("printableStatus")
+            error_list = VM_ERROR_STATUSES.copy()
+            vm_devices = vm.instance.spec.template.spec.domain.devices
+            if vm_devices.gpus:
```

Comment:
no need to save, use directly
```suggestion
            if vm.instance.spec.template.spec.domain.devices.gpus:
```